napari/layers/labels/labels.py

```diff
@@ -159,7 +163,7 @@ def __init__(
         super().__init__(
             data,
             rgb=False,
-            colormap=colormap,
+            colormap=self.random_colormap,
```

Comment:
ideally here we already know what colormap we're using and can pass the actual one (i.e. based off of random, or the color_dict)

Reply:
when the mode is updated a bit later in the initialization, the colormap would be updated anyway